service/connection/ClientState.kt

```diff
@@ -176,6 +176,8 @@ class ClientState constructor(
         return null // TODO
     }
 
+    fun fetchSchema(database: String): String? = _client?.databases()?.get(database)?.schema()
+
```

Comment:
We didn't support this functionality before, added to `ClientState`.

Reply:
it does also fit alongside the delete database functionality. It would also fit in a `DatabaseService` which we don't seem to have? Then all database operations can be moved into there (including create, delete, schema, etc.)
Also, I think `fetchSchema` should go after all the other Database operations in this file.

Reply:
Agreed to the latter. I think these fit in `ClientState` given the functionality for deleting databases also being here.

Renamed to `tryFetchSchema` in line with the pattern of prefixing potentially fallible functions with 'try'.